src/Builder/Request/ProductBuilder.php

```diff
@@ -79,7 +80,15 @@ public function addImages(ProductInterface $product): void
     public function addVariants(ProductInterface $product, ChannelInterface $channel, ?string $localeCode = null): void
     {
         $variants = [];
-        $productVariants = method_exists($product, 'getEnabledVariants') ? $product->getEnabledVariants() : $product->getVariants();
+        if (
+            method_exists($product, 'getEnabledVariants')
+            && $product->getEnabledVariants() instanceof ArrayCollection
```

Comment:
No need to check if it's ArrayCollection as it's already have a return typehint for Collection